sql/src/main/java/io/crate/exceptions/MultiException.java

```diff
@@ -61,7 +67,33 @@ public void printStackTrace(PrintWriter s) {
     @Override
     public String getMessage() {
         StringJoiner joiner = new StringJoiner("\n");
-        exceptions.forEach(e -> joiner.add(e.getMessage()));
+        for (Throwable e : exceptions) {
+            joiner.add(e.getMessage());
+            // Reduce the amount of characters returned to consume less memory
+            // this could escalate on a huge amount of exceptions that are raised by affected shards in the cluster
+            if (joiner.length() > MAX_EXCEPTION_LENGTH) {
+                joiner.add("too much output. output truncated.");
+                break;
+            }
+        }
         return joiner.toString();
     }
+
+    @VisibleForTesting
+    Collection<? extends Throwable> getExceptions() {
```

Comment:
still only used in testing, pls use it at `of` to get a real use-case for this method